[MCLOUD-4910] Escape UC names during data prep

naren-loganathan opened the pull request:
[JIRA](https://databricks.atlassian.net/browse/MCLOUD-4910). This PR aims to get past an issue that occurs when specifying a delta table for training data that has a valid name with hyphens/special characters.

We now escape the table names with backticks.

Testing: `test_format_tablename`

Manual testing: When submitting fine-tuning runs with custom YAMLs pointing to this branch, the logs indicate that delta tables are read without `[INVALID_IDENTIFIER]` errors - even when hyphenated.

NOTE: We always apply escaping (`` `catalog`.`schema`.`tablename` ``). We do not expect the user to escape their own input. We do not handle cases where UC object names contain backticks.

XiaohanZhangCMU reviewed (changes requested):
Some general comments:

1. I thought it is easier just directly format args.delta_table_name in the main function, because you don't need to change any of the rest code.  
2. Can you add a test case where the tablename or schema name already comes in with the right format which has hyphen and escape? 
3. Can you reformat schema as well? I assume it follows the same reformatting rules.

naren-loganathan commented:
Thanks! Addressing these comments:

1. Fixed and verified manually with a custom YAML!
2. It seems like users cannot provide their own escaped UC paths due to validation checks being performed elsewhere. We will not handle names with backticks/escaped names in general. 
3. The expected delta table name is of the form `catalog.schema.tablename`. We escape the catalog, schema and tablename with backticks (i.e. we now do `catalog.schema.tablename` -> ``` `catalog`.`schema`.`tablename` ```).

jjanezhang commented on scripts/data_prep/convert_delta_to_json.py:
```diff
@@ -283,6 +285,24 @@ def download(
 def download_starargs(args: Tuple) -> None:
     return download(*args)
 
+def format_tablename(tablename: str) -> str:
```
```suggestion
def format_tablename(table_name: str) -> str:
```

jjanezhang commented on scripts/data_prep/convert_delta_to_json.py:
```diff
@@ -610,7 +631,7 @@ def fetch_DT(args: Namespace) -> None:
         '--delta_table_name',
         required=True,
-        type=str,
+        type=format_tablename,
```
do you like this better than keeping this as string and then running format_table_name in fetch DT?

naren-loganathan replied:
Now running this within `fetch_DT` to make things clearer.

jjanezhang reviewed (approved):
small nits! otherwise lgtm

XiaohanZhangCMU reviewed (approved):
LGTM. Please test the script a few times with actual tables (w/o hyphens). 

One example is (from a table in staging dogfood):
python scripts/data_prep/convert_delta_to_json.py --delta_table_name main.streaming.random_large_table --json_output_path /tmp/test_json1 --use_serverless True -- xxxxx

naren-loganathan commented:
Was able to get through data prep with `main.streaming.random_large_table`.
